core/src/test/java/com/backblaze/b2/json/B2JsonTest.java

```diff
@@ -1542,6 +1561,80 @@ public void testSerializeUnregisteredUnionSubType() throws B2JsonException {
         B2Json.get().toJson(container);
     }
 
+    @B2Json.union(typeField = "type")
+    @B2Json.defaultForUnknownType(value = "{\"type\": \"a\", \"n\": 5}")
+    private static class UnionWithDefault {
+        public static B2JsonUnionTypeMap getUnionTypeMap() throws B2JsonException {
+            return B2JsonUnionTypeMap
+                    .builder()
+                    .put("a", UnionWithDefaultClassA.class)
+                    .build();
+        }
+    }
+
+    private static class UnionWithDefaultClassA extends UnionWithDefault {
+        @B2Json.required
+        private final int n;
+
+        @B2Json.constructor(params = "n")
+        private UnionWithDefaultClassA(int n) {
+            this.n = n;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+            UnionWithDefaultClassA that = (UnionWithDefaultClassA) o;
+            return n == that.n;
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(n);
+        }
+    }
+
+    @Test
+    public void testUnionWithDefault() throws B2JsonException {
+        assertEquals(
+                new UnionWithDefaultClassA(5), // the default value
+                B2Json.get().fromJson("{\"type\": \"unknown\"}", UnionWithDefault.class)
```

Comment:
add a test with an unknown value that has fields we don't expect?
for instance:

>{
>   "type" : "person",
>   "age" : 47,
>   "extra" : { "foo" : "bar" }
>}

this will make sure we can properly ignore unknown stuff inside the unknown type.

Reply:
Good catch!  It didn't work.  Added test, then added fix to not throw about an unknown field name in a union until the type is extracted, checked, and the default value is used if it's an unknown type.  Then it throws about an unknown field.